src/verify.c

```diff
@@ -474,6 +478,29 @@ static void add_missing_files(struct manifest *official_manifest)
 		if ((ret != 0) || hash_needs_work(file, local.hash)) {
 			file_not_replaced_count++;
 			printf("\n\tnot fixed\n");
+
+			if (cmdline_option_install && no_freespace_flag == false) {
+				fs_free = get_available_space(path_prefix);
+				string_or_die(&original, "%s/staged/%s", state_dir, file->hash);
```

Comment:
This is an interesting thing to check against, since it *could* be that someone is installing the OS to the same partition on which this file lives (installing to a directory in the same rootfs for instance). In this case it could be possible that the state directory is on the same file system and if the user got to this point it *could* still pass even if there is no more room on the device because we try to hardlink everything if we can.

This is fine though because these are just warnings, the write *did* fail, and the disk is getting full anyways.